src/pages/SignUp/SignUp.tsx

```diff
@@ -0,0 +1,275 @@
+import React, { useEffect } from "react";
+import { Button, Card, Container, Form, FormGroup } from "react-bootstrap";
+import { useForm, useFormState } from "react-hook-form";
+import { useLocation, useNavigate } from "react-router-dom";
+import { toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+import {
+  capitalLetterPattern,
+  emailpattern,
+  lowerLetterPattern,
+  numbersPattern,
+  specialCharacterPattern,
+} from "../../constants";
+import "./SignUp.scss";
+
+function SignUp() {
+  const location = useLocation();
+  useEffect(() => {
+    window.scrollTo(0, 0);
+  }, [location]);
+
+  const navigate = useNavigate();
+  const {
+    register,
+    handleSubmit,
+    control,
+    watch,
+    formState: { errors },
+  } = useForm({
+    mode: "onChange",
+    reValidateMode: "onChange",
+    criteriaMode: "all",
+  });
+
+  const { dirtyFields } = useFormState({
+    control,
+  });
+  const watchPassword = watch("password");
+
+  useEffect(() => {
+    document.title = "Xtreme Cars | Sign Up";
+  }, []);
+  const onSubmit = (formData: any) => {
+    toast.success("Profile Created Successfully");
+    navigate("/home");
+  };
+
+  return (
+    <>
+      <Container>
+        <h3>Sign Up</h3>
+        <Card className="form-content">
+          <Form onSubmit={handleSubmit(onSubmit)} autoComplete="off">
+            <Form.Group
+              className="mb-3 form-group-container"
+              controlId="formGroupName"
+            >
+              <Form.Label>Name</Form.Label>
+              <Form.Control
+                type="text"
+                placeholder="Enter name..."
+                {...register("name", { required: "Name is required" })}
+              />
+              {errors.name && (
+                <p className="text-danger" data-testid="name-error">
+                  {errors.name.message}
+                </p>
+              )}
+            </Form.Group>
+            <Form.Group
+              className="mb-3 form-group-container"
+              controlId="formGroupContact"
+            >
+              <Form.Label>Mobile Number</Form.Label>
+              <Form.Control
+                type="tel"
+                placeholder="Enter contact..."
+                {...register("contact", {
+                  required: "Mobile Number is required",
+                  minLength: {
+                    value: 10,
+                    message: "Mobile number cannot be less than 10 digits",
+                  },
+                  maxLength: {
+                    value: 10,
+                    message: "Mobile number cannot be more than 10 digits",
+                  },
```

Comment:
Not checking for pattern, like if the mobile number 10 digit and starts with 1-5 will it be valid?
if invalid please add that validation along with message "Invalid Mobile number"